cppcache/integration/test/PartitionRegionOpsTest.cpp

```diff
@@ -191,74 +237,85 @@ void getPartitionedRegionWithRedundancyServerGoesDown(bool singleHop) {
  * In this test case we verify that in a partition region with redundancy
- * when one server goes down, all gets are still served.
+ * when one server goes down, all puts and gets are still served.
+ *
  * Single-hop is enabled in the client.
+ *
  * It can be observed in the logs that when one of the server goes down
  * the bucketServerLocations for that server are removed from the
  * client metadata.
  */
-TEST(PartitionRegionOpsTest,
-     getPartitionedRegionWithRedundancyServerGoesDownSingleHop) {
+TEST(PartitionRegionWithRedundancyTest, putgetWithSingleHop) {
   removeLogFromPreviousExecution();
-  getPartitionedRegionWithRedundancyServerGoesDown(true);
+  bool useSingleHop = true;
+  putget(useSingleHop);
   verifyMetadataWasRemovedAtFirstError();
 }
 
 /**
  * In this test case we verify that in a partition region with redundancy
- * when one server goes down, all puts are still served.
+ * when one server goes down, all putAlls and getAlls are still served.
+ *
  * Single-hop is enabled in the client.
+ *
  * It can be observed in the logs that when one of the server goes down
  * the bucketServerLocations for that server are removed from the
  * client metadata.
- * When the server is brought back again, the meta data is refreshed
- * after putting again values.
  */
-TEST(PartitionRegionOpsTest,
-     putPartitionedRegionWithRedundancyServerGoesDownSingleHop) {
+TEST(PartitionRegionWithRedundancyTest, putAllgetAllWithSingleHop) {
   removeLogFromPreviousExecution();
-  putPartitionedRegionWithRedundancyServerGoesDown(true);
+  bool useSingleHop = true;
+  putAllgetAll(useSingleHop);
   verifyMetadataWasRemovedAtFirstError();
 }
 
 /**
  * In this test case we verify that in a partition region with redundancy
- * when one server goes down, all gets are still served.
+ * when one server goes down, all puts and gets are still served.
+ *
  * Single hop is not enabled in the client.
+ *
+ * It can be observed in the logs that no metadata was requested.
  */
-TEST(PartitionRegionOpsTest,
-     getPartitionedRegionWithRedundancyServerGoesDownNoSingleHop) {
-  getPartitionedRegionWithRedundancyServerGoesDown(false);
+TEST(PartitionRegionWithRedundancyTest, putgetWithoutSingleHop) {
+  removeLogFromPreviousExecution();
+  bool useSingleHop = false;
+  putget(useSingleHop);
+  verifyNoMetaData();
 }
 
 /**
  * In this test case we verify that in a partition region with redundancy
- * when one server goes down, all puts are still served.
- * Single-hop is not enabled in the client.
+ * when one server goes down, all putAlls and getAlls are still served.
+ *
+ * Single hop is not enabled in the client.
+ *
+ * It can be observed in the logs that no metadata was requested.
  */
-TEST(PartitionRegionOpsTest,
-     putPartitionedRegionWithRedundancyServerGoesDownNoSingleHop) {
-  putPartitionedRegionWithRedundancyServerGoesDown(false);
+TEST(PartitionRegionWithRedundancyTest, putAllgetAllWithoutSingleHop) {
+  removeLogFromPreviousExecution();
```

Comment:
Why are logs involved here at all. Testing base on logs should be avoided at all cost.

Reply:
Investigating adding ClientMetaDataHelper to improve testability of ClientMetaData.

Reply:
Removed reliance on logfile parsing by leveraging CacheImpl::getAndResetNetworkHopFlag(), which was already exported.